startup.py

```diff
@@ -0,0 +1,253 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import uuid
+import imp
+import sgtk
+
+from sgtk.platform import SoftwareLauncher, SoftwareVersion, LaunchInformation
+
+
+class NukeLauncher(SoftwareLauncher):
+    """
+    Handles launching Nuke executables. Automatically starts up a tk-nuke
+    engine with the current context in the new session of Nuke.
+    """
+
+    # Named regex strings to insert into the executable template paths when
+    # matching against supplied versions and products. Similar to the glob
+    # strings, these allow us to alter the regex matching for any of the
+    # variable components of the path in one place
+
+    COMPONENT_REGEX_LOOKUP = {
+        "version": r"(?P<version>[\d.v]+)",
```

Comment:
Should we simplify to not use the `?P<>` syntax now that we can disguise it via the helper method?